### DISCUSSION TITLE
Apollo client usequery with variable

### DISCUSSION THREAD

**Discussion by ChrisBagge** [MichalLytek/type-graphql#1108] [Q&A] [answered]
This seems pretty straight forward in the apollo client documentation but i cant figure out how to pass a parameter using usequery. I am following the apollo client docs for typescript and i see a parameter that appears the root level in Apollo dev tools but it does not get passed to my query. I am using the simple recipe example for my backend with a react app pm frontend. I can query just find if i hardcode a value for recipe id but the $ variable will not work for me. 

Can anyone shed some light on this please?

thanks,
Chris

**Comment by MichalLytek** [comment, marked as answer]
I'm not sure if someone in TypeGraphQL discussions will help you setup Apollo Client 😛

**Reply by ChrisBagge** [reply]
I hope you are wrong  (insert smiley face here) Here is my query. Do you see anything wrong with that? I see the variable in the apollo dev tools in the browser but the query does not return anything.

  query blog($id: number) {
    blog(id: $id) {
      title
      body
      author
      id
    }
  }

This here does return

query blog {
    blog(id: 1) {
      title
      body
      author
      id
    }
  }

As soon as I add in that variable it just stops working
